File: Swift-C++-IR-Comparison/test/structsAndClasses/Swift/Classes/InstanceMethodCounter.swift
//
//  File.swift
//
//
//  Created by Jan-Niclas Strüwer on 26.10.22.
//

import Foundation

@main
struct MyMain {

    static func main() {
        let c = Counter()
        c.increment()
        c.increment(by: 20)
        c.reset()
        c.increment()
        let currentCount = c.count
    }
}
class Counter {
    var count = 0
    func increment() {
        count += 1
    }
    func increment(by amount: Int) {
        count += amount
    }
    func reset() {
        count = 0
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-0/InstanceMethodCounter.cpp

#include <iostream>

class Counter
{
    int count = 0;
public:
    int get_count() { return count; }
    void increment() { count += 1; }
    void increment(int amount) { count += amount; }
    void reset() { count = 0; }
};

int main()
{
    Counter c;
    c.increment();
    c.increment(20);
    c.reset();
    c.increment();
    int currentCount = c.get_count();
}